makefile

```diff
@@ -9,16 +9,20 @@
 #
 ###############################################################################
 
-BUILD_PATH    := build
-TARGETS_PATH  := targets
-APPS_PATH     := apps
-APP_OUT_PATH  := ../$(BUILD_PATH)/bin/$(APPS_PATH)/$(APP)
-OBJ_PATH      := ../$(BUILD_PATH)/obj
-TARGET_PATH   := ../$(TARGETS_PATH)/$(TARGET)
-APP_PATH      := ../$(APPS_PATH)/$(APP)
-LIB_PATH      := ../lib
-MBED_PATH     := ../mbed-os
-CONFIG_PATH   := ../config
+BUILD_FOLDER   := build
+BIN_FOLDER     := bin
+TARGETS_FOLDER := targets
+APPS_FOLDER    := apps
+OBJS_FOLDER    := obj
+BIN_OUT_PATH   := ../$(BUILD_FOLDER)/$(APPS_FOLDER)/$(BIN_FOLDER)/$(APP)
+OBJ_PATH       := ../$(BUILD_FOLDER)/$(OBJS_FOLDER)
+TARGET_PATH    := ../$(TARGETS_FOLDER)/$(TARGET)
+APP_PATH       := ../$(APPS_FOLDER)/$(APP)
+LIB_PATH       := ../lib
+MBED_PATH      := ../mbed-os
+CONFIG_PATH    := ../config
+
+LAST_BOARD_TARGET := $(shell cat $(BUILD_FOLDER)/LAST_BOARD_TARGET 2>/dev/null)# ignores non-existant file error
```

Comment:
there is a file function but that requires make 4.2. Ubuntu 18 only ships make 4.1

Reply:
also this assumes u have cat, which is part of gnu. assumption is that if u have make, which is also a part of gnu, you should have cat. ex. cygwin or mingw for windows. or wsl. or ubuntu. 

sidenote: do we even still support mac? or non-wsl windows?

Reply:
Mac is supported, Windows without WSL is not.

Reply:
Ok. I dont have a mac to test, but I dont see why my changes would break it

Reply:
I don't see anything either